### ISSUE TITLE
False positive on next() within generator

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#2158]
### Steps to reproduce

Run pylint on the following code:

```python
from itertools import count

def generator_using_next():
    counter = count()
    number = next(counter)
    yield number * 2
```

### Current behavior

The following issue is reported:

```
R:  5,13: Do not raise StopIteration in generator, use return statement instead (stop-iteration-return)
```

### Expected behavior

I don't think anything has to be reported here.

In general, `next()` might raise `StopIteration`, but not in every case, since there are infinite generators like `count()` that I used in the example. So stating that `StopIteration` is raised is inaccurate.

In cases where `next()` can raise `StopIteration`, that could be considered a somewhat opaque execution flow, so there is an argument to be made for avoiding that. However, the solution would then not be just the use of a return statement, since the raising of `StopIteration` is conditional. Instead, it would have to be something like catching `StopIteration` and then returning from the `except:` clause, which is a block of extra code that I'm not sure is worth it.

### pylint --version output

```
No config file found, using default configuration
pylint 1.9.1, 
astroid 1.6.4
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```

**Comment by mthuurne** [comment]
Thanks for the very quick fix!

However, the original code that triggered the false positive doesn't actually use `itertools.count`, it uses a custom iterator containing a `while True:` loop. So it would be good to have infinite iterator detection that works for custom (defined outside the standard library) iterators as well. Should I open a separate issue for that feature request or is it better to re-open this one?
